server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -112,6 +113,63 @@ public NodeIndicesStats(CommonStats oldStats, Map<Index, List<IndexShardStats>>
         }
     }
 
+    public NodeIndicesStats(
```

Comment:
Created another constructor as `NodeIndicesStats` is a [Public API](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/NodeIndicesStats.java#L75). 

The whole point of this new constructor is to:
- Firstly, build the `CommonStats` as it would do in previous versions as well (This is the node level aggregations of the stats)
- Then depending on the level passed, we decide to either create `statsByIndex` (Index level aggregations) or just return `statsByShards` as we would do normally.

The ser/de accordingly handles the `statsByIndex` and `statsByShards` if they are null or not.